runtime/src/accounts_db.rs

```diff
@@ -124,6 +129,97 @@ impl Versioned for (u64, AccountInfo) {
     }
 }
 
+pub enum LoadedAccountAccessor<'a> {
+    Stored(Option<(Arc<AccountStorageEntry>, usize)>),
+    Cached((&'a AccountsCache, Slot, &'a Pubkey)),
+}
+
+impl<'a> LoadedAccountAccessor<'a> {
+    fn get_loaded_account(&self) -> Option<LoadedAccount> {
```

Comment:
maybe this function can be called inside get_account_accessor_from_cache_or_storage? it seems that all callers of get_account_accessor_from_cache_or_storage immediately calls this.

Reply:
hehe so the issue in `get_account_accessor_from_cache_or_storage` is a familiar one 😃. We want to return both:
1. The result of `storage_entry = get_account_storage_entry()`
2. `storage_entry..get_stored_account_meta(*offset)` for some offset, which is a reference to the data inside `storage_entry`

This is creates the old self-referential issue. My appraoch here was to return 1) in  `get_account_accessor_from_cache_or_storage()`, and then while thats in scope, return 2) in `get_loaded_account()`